### PR TITLE
[mac] add option to change direct/indirect TX retry number

### PR DESCRIPTION
**Pull request by e-rk**
This PR adds an option to set the number of direct/indirect number of TX retries on the MAC layer.

The PR covers:
* Public API
* NCP properties
* CLI commands

### CODE REVIEW

File: src/core/config/openthread-core-config-check.h [hunk 1]
```diff
@@ -441,4 +441,9 @@
     "OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_METRICS_ENABLE was replaced by OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE."
 #endif
 
+#ifdef OPENTHREAD_CONFIG_MAC_MAX_FRAME_RETRIES_DIRECT
```
Comment (LuDuda) [inline src/core/config/openthread-core-config-check.h]: Can we do the same for `OPENTHREAD_CONFIG_MAC_MAX_FRAME_RETRIES_INDIRECT` define?

File: src/core/mac/mac.hpp [hunk 2]
```diff
@@ -425,6 +426,43 @@ class Mac : public InstanceLocator
...
+     *
+     * @returns The maximum number of retries during indirect transmission.
+     *
+     */
+    uint8_t GetMaxFrameRetriesIndirect(void) const { return mMaxFrameRetriesIndirect; }
+
+    /**
+     * This method sets the maximum number of frame retries during indirect transmission.
+     *
+     * @param[in]  aMaxFrameRetriesIndirect  The maximum number of retries induring indirect transmission.
```
Comment (LuDuda) [inline src/core/mac/mac.hpp]: nit: induring -> during

File: src/core/config/openthread-core-config-check.h (hunk 1 above)
Comment (e-rk) [reply src/core/config/openthread-core-config-check.h]: I forgot about that one. Thanks.

**Comment by LuDuda** [comment]
Any chance to get some initial feedback about this PR?

File: src/ncp/spinel.h [hunk 3]
```diff
@@ -1927,6 +1927,22 @@ typedef enum
      */
     SPINEL_PROP_MAC_CCA_FAILURE_RATE = SPINEL_PROP_MAC_EXT__BEGIN + 9,
 
+    /// MAC Max direct retry number
+    /** Format: `C`
+     *
+     * The maximum (user-specified) number of direct frame transmission retries.
+     *
+     */
+    SPINEL_PROP_MAC_MAX_RETRY_NUMBER_DIRECT = SPINEL_PROP_MAC_EXT__BEGIN + 10,
```
Comment (abtink) [inline src/ncp/spinel.h]: nit: can you please add support for new porp definition in spinel.c `spinel_prop_key_to_cstr()` function? Thanks

File: src/ncp/spinel.h [hunk 4]
```diff
@@ -1927,6 +1927,22 @@ typedef enum
...
+
+    /// MAC Max indirect retry number
+    /** Format: `C`
+     *
```
Comment (abtink) [inline src/ncp/spinel.h]: nit: how about adding  `Required capability: CAP_CONFIG_FTD` to document it can be used on FTD only?

**Comment by e-rk** [comment]
Thanks @abtink.
In the meantime we have considered changing the `macretries` name into `mac retries` instead. That way, the `mac` related commands would be under one umbrella prefix. Would that be okay with you?

**Comment by jwhui** [comment]
@e-rk, having a `mac` prefix works for me. We could also move `macfilter` to `mac filter` in a later PR.

**Pull request merged**